vmdb/app/models/ems_kubernetes.rb

```diff
@@ -35,27 +42,33 @@ def api_endpoint
     self.class.raw_api_endpoint(hostname, port)
   end
 
-  def connect(_options = {})
-    self.class.raw_connect(hostname, port)
-  end
+  def connect(options = {})
+    hostname = options[:hostname] || self.address
+    port     = options[:port]     || self.port
 
-  def self.event_monitor_class
-    MiqEventCatcherKubernetes
+    self.class.raw_connect(hostname, port)
   end
 
-  def authentication_check
+  def verify_credentials(auth_type = nil, options = {})
     # TODO: support real authentication using certificates
-    [true, ""]
-  end
+    options = options.merge(:auth_type => auth_type)
 
-  def verify_credentials(_auth_type = nil, _options = {})
-    # TODO: support real authentication using certificates
-    true
+    with_provider_connection(options, &:api_valid?)
```

Comment:
Validation here should be a check that remote endpoint supports the API that we know about (v1beta3 / future v1.0).
I'll send a separate patch about that.